### PR TITLE
Making cypress on rails more generic

### PR DESCRIPTION
**Pull request by grantspeelman**
cc @derfred @darioghilardi
This more of a generic PR for a discussion than for actually review, considering i went totally crazy on the code 😄 

I tested out using cypress-on-rails and really like what it does, at the heart of it I like the middleware approach. 
I have rewritten most of the code in attempt to make it more generic and flexible:
* removed scenario banks and just load the ruby files (can still do scenario banks but it becomes user code instead of gem code)
* removed the specifics of rspec and database_cleaner yet you can still use because it's just ruby (the generator creates files for it)
* removed the runner (does not work with docker nicely and restarting rails means restarting both cypress and rails)
* other stuff 😄 

Here is the [middleware](https://github.com/grantspeelman/cypress-on-rails/blob/master/lib/cypress/middleware.rb) which i feel is the main bit.

[README](https://github.com/grantspeelman/cypress-on-rails/blob/ad4d89358400e72cdac353efbc237bd81f7f27ea/README.md) maybe worth reading.

So basically want to know is it worth attempting to get these changes into the code or has it deviated too much?

### CODE REVIEW

**Comment by grantspeelman** [comment]
@derfred awesome, glad you like what you see. 
Sorry just pushing on with it in the meantime because to maybe start using it at my office.
Let me know the ideas you like and I can look at making it more consumable PR 😄

**Comment by grantspeelman** [comment]
@derfred thank you very much for the effort. Think I am just going to release this as a new gem, hope thats fine.

**Comment by derfred** [comment]
@grantspeelman Sorry I did not get around to reviewing your changes. Of course it is find if you release this as a new gem. By now you have done a lot more work on this than me.

If you are interested I can transfer the ownership of the name on rubygems to you.

**Comment by grantspeelman** [comment]
thanks @derfred 👍 
sure we transfer the name on rubygems, what do I have to do to make this happen?

**Comment by derfred** [comment]
Let's connect via email. send me a message to ich at derfred.com
